File: 1828 B. Permutation Swap/Solution.py
# 319951756	        May/17/2025 05:00UTC+6	        Krishno_MD	        1828B - Permutation Swap	        PyPy 3-64	        Accepted	        218 ms	        16900 KB

# Krishno_Mondol_KUET_CSE
from math import gcd
from functools import reduce
def solution():
    n = int(input())
    print(reduce(gcd,(abs(v-i-1) for i,v in enumerate(map(int,input().split())))))

test = int(input())
for i in range(test):
    solution()

File: 1828 B. Permutation Swap/Solution.cpp
// 319951723	        May/17/2025 04:58UTC+6	        Krishno_MD	        1828B - Permutation Swap	        C++20 (GCC 13-64)	        Accepted	        249 ms	        100 KB

#include<bits/stdc++.h>
using namespace std;
//Krishno_Mondol_KUET_CSE 
#define ll long long 
int gcd(int a, int b){return b==0 ? a : gcd(b, a%b);}
void solve(){
    int n,anss = 0; cin>>n;
    for(int i=1; i<=n; i++){int x; cin>>x; anss = gcd(anss,abs(x-i));}
    cout<<anss<<endl;
}
int main(){ int t; cin>>t; while(t--){ solve(); } return 0;}
